### PR TITLE
Feature/990 loading error states tidy

### PR DESCRIPTION
**Pull request by chris-lawton**
Based off `feature/877-note-polling`

- Adds loading, error and empty states to app `DetailView` component
- Adds single loading screen when page is first hit
- Styles listing error messages

### CODE REVIEW

File: opentech/static_src/src/app/src/components/DetailView/index.js [hunk 1]
```diff
@@ -26,13 +30,12 @@ class DetailView extends Component {
     }
 
     render() {
-        const { listing, submissionID } = this.props;
-
-        const activeSubmision = !!submissionID;
+        const { listing, isLoading, error, isEmpty, showSubmision } = this.props;
+        const isError = Boolean(error);
```
Comment (todd-dembrey) [inline opentech/static_src/src/app/src/components/DetailView/index.js]: Maybe we want `isError` and `errorMessage` separated

File: opentech/static_src/src/app/src/containers/GroupByRoundDetailView.js [hunk 2]
```diff
@@ -1,18 +1,60 @@
...
+        const activeSubmision = !!submissionID;
+
         return (
-            <DetailView listing={listing} />
+            <DetailView
+                isEmpty={isEmpty}
+                listing={listing}
+                isLoading={isLoading}
+                showSubmision={activeSubmision}
+                error={isErrored ? 'Fetching failed.' : undefined}
```
Comment (todd-dembrey) [inline opentech/static_src/src/app/src/containers/GroupByRoundDetailView.js]: Can we not grab an error message from the store? 

Equally - pass in `isErrored` to be explicit.

so we can do:
```
    isErrored={ isErrored },
    errorMessage= { errorMessageFromStore || 'Fetching failed.' }
```

File: opentech/static_src/src/app/src/containers/GroupByStatusDetailView.js [hunk 3]
```diff
@@ -1,13 +1,52 @@
...
         const listing = <ByStatusListing />;
+        const { round, error, submissions, submissionID } = this.props;
+        const isLoading = !round || (round && (round.isFetching || round.submissions.isFetching))
+        const isEmpty = submissions.length === 0;
+        const activeSubmision = !!submissionID;
+
         return (
-            <DetailView listing={listing} />
+            <DetailView
+                error={error}
```
Comment (todd-dembrey) [inline opentech/static_src/src/app/src/containers/GroupByStatusDetailView.js]: As above

File: opentech/static_src/src/app/src/components/DetailView/index.js [hunk 4]
```diff
@@ -52,24 +55,36 @@ class DetailView extends Component {
                 </div>
             )
         } else {
+            if (isLoading) {
+                return (
+                    <LoadingPanel />
+                )
+            } else if (isError) {
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply opentech/static_src/src/app/src/components/DetailView/index.js]: Yup, agreed. Moved up to include mobile now

File: opentech/static_src/src/app/src/components/GroupedListing/index.js [hunk 5]
```diff
@@ -139,8 +139,6 @@ export default class GroupedListing extends React.Component {
                 <div className="grouped-listing__dropdown" ref={(ref) => this.dropdownContainer = ref}>
                     {!error && !isLoading &&
                         <ListingDropdown
-                            error={error}
-                            isLoading={isLoading}
                             listRef={this.listRef}
```
[the remark this answers is not in the document]
Comment (chris-lawton) [reply opentech/static_src/src/app/src/components/GroupedListing/index.js]: Is this not handled at the `DetailView` level now though or is there an instance where you can get past that but the dropdown will still be empty?

File: opentech/static_src/src/app/src/components/DetailView/index.js (hunk 1 above)
Comment (chris-lawton) [reply opentech/static_src/src/app/src/components/DetailView/index.js]: 👍

File: opentech/static_src/src/app/src/containers/GroupByRoundDetailView.js (hunk 2 above)
Comment (chris-lawton) [reply opentech/static_src/src/app/src/containers/GroupByRoundDetailView.js]: Yeah, I prefer this. I guess we should now follow suit in `ByRoundListing`? https://github.com/OpenTechFund/opentech.fund/blob/master/opentech/static_src/src/app/src/containers/ByRoundListing.js#L75

Comment (todd-dembrey) [reply opentech/static_src/src/app/src/containers/GroupByRoundDetailView.js]: Yes, lets be consistent.

**Pull request merged**